manifests/image.pp

```diff
@@ -74,10 +75,17 @@
     $image_find    = "${docker_command} images | cut -d ' ' -f 1 | egrep '^(docker\\.io/)?${image}$'"
   }
 
+  if $build_args {
+    validate_array($build_args)
+    $build_args_string = join ( regsubst ($build_args, '^(.*)$', '--build-arg \1 '), '')
+  } else {
+    $build_args_string = ''
+  }
+
   if $docker_dir {
-    $image_install = "${docker_command} build -t ${image_arg} ${docker_dir}"
+    $image_install = "${docker_command} build ${build_args_string}-t ${image_arg} ${docker_dir}"
```

Comment:
I'm dealing with the space when building the 'build_args_string', i don't want to put an extra space when build_args_string is empty. Do have a better way of dealing with that?

Still i saw the Ruby 1.8.7 build failing because of the rake version is that something i should try to fix on the Gemfile?